src/S3/PostObjectV4.php

```diff
@@ -0,0 +1,214 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Credentials\CredentialsInterface;
+use GuzzleHttp\Psr7\Uri;
+use Aws\Signature\SignatureTrait;
+use Aws\Common\Enum\DateFormat;
+
+/**
+ * Encapsulates the logic for getting the data for an S3 object POST upload form
+ *
+ * @link http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html
+ * @link http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
+ */
+class PostObjectV4
+{
+    use SignatureTrait;
+
+    const ISO8601_BASIC = 'Ymd\THis\Z';
+
+    private $client;
+    private $bucket;
+    private $formAttributes;
+    private $formInputs = [];
+    private $jsonPolicy;
+    private $credentials;
+
+    /**
+     * Constructs the PostObject.
+     *
+     * The options array accepts the following keys:
+     * @link http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+     *
+     * @param S3ClientInterface $client     Client used with the POST object
+     * @param string            $bucket     Bucket to use
+     * @param array             $options    Policy condition options
+     * @param string            $expiration Upload expiration date. By
```

Comment:
This should be polymorphic. Usually the SDK will take an integer, `DateTime`, or string when a date value is needed.